datalad/core/distributed/clone.py

```diff
@@ -1311,6 +1321,28 @@ def decode_source_spec(spec, cfg=None):
         # if we still have no good idea on where a dataset could be cloned to if no
         # path was given, do something similar to git clone and derive the path from
         # the source
-        props['default_destpath'] = _get_installationpath_from_url(props['giturl'])
+        props['default_destpath'] = _get_installationpath_from_url(
+            props['giturl'])
 
     return props
+
+
+def _add_hint_on_misbuilt_urls(urls, error_msgs):
+    """If cloning failed, but a KeyError was caught in the generation of a
+    submodule clone candidate URL, the misbuilt submodule URL template is
+    reported to the user
+    :param urls: dict; contains a submodule and unmatched URL template
+    :return: error_msgs; OrderedDict of encountered cloning errors
+    """
+    # import pdb; pdb.set_trace()
+    from datalad.ui import ui
+    from datalad.support import ansi_colors
+    intro = ansi_colors.color_word(
+        "Hint: A configuration for custom subdataset clone candidates exists, "
+        "but no key matched the template: ",
```

Comment:
Might be that some keys did match. 
```suggestion
        "but some key(s) used in the template were missing: ",
```
and unfortunately we don't channel to user here which key

Reply:
this is better, I added it. I'm also reporting the key now.